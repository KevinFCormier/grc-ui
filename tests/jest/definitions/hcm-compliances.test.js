/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'
import {
  createPolicyLink,
  createCompliancePolicyLink,
  getStatus,
  getStatusIcon,
  getStatusIconForPolicy,
} from '../../../src-web/definitions/hcm-compliances'

describe('hcm-compliances - createPolicyLink', () => {
  it('should return valid policy link', () => {
    const item = {
      metadata: {
        name: 'testName',
        namespace: 'testNamespace'
      }
    }
    expect(createPolicyLink(item)).toMatchSnapshot()
  })
})

describe('hcm-compliances - createCompliancePolicyLink', () => {
  it('should return valid compliance policy link', () => {
    const item = {
      cluster: 'clusterName',
      complianceName: 'testComplianceName',
      complianceNamespace: 'testComplianceNamespace',
      metadata: {
        name: 'testName',
      }
    }
    expect(createCompliancePolicyLink(item)).toMatchSnapshot()
  })
})

describe('hcm-compliances - getStatus', () => {
  it('should return compliant status', () => {
    const item = {
      status: 'compliant'
    }
    expect(getStatus(item, 'en-US')).toMatchSnapshot()
  })
  it('should return "-"', () => {
    const item = {
    }
    expect(getStatus(item, 'en-US')).toBe('-')
  })
})

describe('hcm-compliances - getStatusIcon', () => {
  it('should return compliant status', () => {
    const item = {
      compliant: 'compliant'
    }
    expect(getStatusIcon(item, 'en-US')).toMatchSnapshot()
  })
  it('should return noncompliant status', () => {
    const item = {
      compliant: 'noncompliant'
    }
    expect(getStatusIcon(item, 'en-US')).toMatchSnapshot()
  })
  it('should return "-"', () => {
    const item = {
    }
    expect(getStatusIcon(item, 'en-US')).toBe('-')
  })
})

describe('hcm-compliances - getStatusIconForPolicy', () => {
  it('should return compliant status', () => {
    const item = {
      status: 'compliant'
    }
    expect(getStatusIconForPolicy(item, 'en-US')).toMatchSnapshot()
  })
  it('should return noncompliant status', () => {
    const item = {
      status: 'noncompliant'
    }
    expect(getStatusIconForPolicy(item, 'en-US')).toMatchSnapshot()
  })
  it('should return "-"', () => {
    const item = {
    }
    expect(getStatusIconForPolicy(item, 'en-US')).toBe('-')
  })
})
