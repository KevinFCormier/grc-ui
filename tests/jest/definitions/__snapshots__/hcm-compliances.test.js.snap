// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hcm-compliances - createCompliancePolicyLink should return valid compliance policy link 1`] = `"-"`;

exports[`hcm-compliances - createPolicyLink should return valid policy link 1`] = `
<Link
  replace={false}
  to="undefined/policies/undefined/undefined/compliancePolicy/undefined"
/>
`;

exports[`hcm-compliances - getStatus should return compliant status 1`] = `
<StatusField
  status="ok"
  text="compliant"
/>
`;

exports[`hcm-compliances - getStatusIcon should return compliant status 1`] = `
<StatusField
  status="ok"
  text="compliant"
/>
`;

exports[`hcm-compliances - getStatusIcon should return noncompliant status 1`] = `
<StatusField
  status="critical"
  text="not compliant"
/>
`;

exports[`hcm-compliances - getStatusIconForPolicy should return compliant status 1`] = `
<div
  className="compliance-table-status"
>
  <Icon
    className="table-status__compliant"
    description=""
    fillRule="evenodd"
    name="icon--checkmark--glyph"
    role="img"
  />
</div>
`;

exports[`hcm-compliances - getStatusIconForPolicy should return noncompliant status 1`] = `
<div
  className="compliance-table-status"
>
  <Icon
    className="table-status__not_compliant"
    description=""
    fillRule="evenodd"
    name="icon--error--glyph"
    role="img"
  />
</div>
`;
